core/hash/compare_by_identity_spec.rb

```diff
@@ -44,9 +45,9 @@ def o.hash; 123; end
     :bar.should equal(:bar)
     @idh[:bar] = :e
     @idh[:bar] = :f
-    "bar".should_not equal('bar')
-    @idh["bar"] = :g
-    @idh["bar".dup] = :h
+    ["bar"].should_not equal(['bar'])
```

Comment:
Since this now uses an Array, I would rather keep it simple with an Array of integers so String semantics are not involved at all.